src/test/scala/com/microsoft/ml/spark/lightgbm/split1/VerifyLightGBMClassifier.scala

```diff
@@ -387,6 +387,72 @@ class VerifyLightGBMClassifier extends Benchmarks with EstimatorFuzzing[LightGBM
     assert(model != null)
   }
 
+  test("Verify LightGBM Classifier won't get stuck on unbalanced classes in multiclass classification") {
+    assume(!isWindows)
+    // Increment port index
+    portIndex += numPartitions
+    val fileName = "BreastTissue.csv"
+    val labelColumnName = "Class"
+    val fileLocation = DatasetUtils.multiclassTrainFile(fileName).toString
+    val dataset = readCSV(fileName, fileLocation).repartition(numPartitions)
+    val featuresColumn = "_features"
+    val featurizer = LightGBMUtils.featurizeData(dataset, labelColumnName, featuresColumn)
+    val predCol = "pred"
+    val tmpTrainData = featurizer.transform(dataset)
+    val labelizer = new ValueIndexer().setInputCol(labelColumnName).setOutputCol(labelColumnName).fit(tmpTrainData)
+    val labelizedData = labelizer.transform(tmpTrainData)
+    val lgbm = new LightGBMClassifier()
+      .setLabelCol(labelColumnName)
+      .setFeaturesCol(featuresColumn)
+      .setPredictionCol(predCol)
+      .setDefaultListenPort(LightGBMConstants.DefaultLocalListenPort + portIndex)
+      .setObjective(multiclassObject)
+
+    val infoSchema = new StructType()
+      .add(labelColumnName, IntegerType).add(featuresColumn, org.apache.spark.ml.linalg.SQLDataTypes.VectorType)
+    val infoEnc = RowEncoder(infoSchema)
+    val trainData = labelizedData.select(labelColumnName, featuresColumn)
+      .mapPartitions(iter => {
+        val ctx = TaskContext.get
+        val partId = ctx.partitionId
+        // Remove all instances of some classes
+        if (partId != 0) {
+          iter.flatMap(row => {
```

Comment:
abstract the duplicated code

Reply:
created an issue since there seems to be a lot of boilerplate code in test file that could be refactored out https://github.com/Azure/mmlspark/issues/654